packages/eslint-plugin-wpcalypso/lib/rules/i18n-unlocalized-url.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * @file Disallow using unlocalized URL strings
+ * @author Automattic
+ * @copyright 2023 Automattic. All rights reserved.
+ * See LICENSE.md file in root directory for full license.
+ */
+
+//------------------------------------------------------------------------------
+// Constants
+//------------------------------------------------------------------------------
+
+const LOCALIZABLE_URLS = [
+	'apps.wordpress.com',
+	'automattic.com/cookies(/|$)',
+	'automattic.com/privacy(/|$)',
+	'wordpress.com/blog(/|$)',
+	'wordpress.com/forums(/|$)',
+	'wordpress.com/go(/|$)',
+	'wordpress.com/support(/|$)',
+	'wordpress.com/tos(/|$)',
+];
+
+//------------------------------------------------------------------------------
+// Helper Functions
+//------------------------------------------------------------------------------
+
+const getCallee = require( '../util/get-callee' );
+
+/**
+ * Check whether the provided string is localizable string.
+ *
+ * @param   {string}  string String to be tested
+ * @returns {boolean}
+ */
+function isLocalizableUrlString( string ) {
+	return LOCALIZABLE_URLS.some( ( url ) =>
+		new RegExp( `^(https?:)?//${ url.replace( '.', '\\.' ) }`, 'i' ).test( string )
```

Comment:
It may be safer to escape all regex characters, otherwise someone could add a URL with another special character and break the regex. 

I'd suggest either using a custom version like:

```
function escapeRegex(string) {
    return string.replace(/[/\-\\^$*+?.()|[\]{}]/g, '\\$&');
}
```

or if you **really have to**, use the Lodash `escapeRegExp` function.

Reply:
I ended up using regex objects directly in the `LOCALIZABLE_URLS` definition.

Reply:
Looks good and the added comments make it extra readable - thanks!